presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -1003,6 +1010,30 @@ public synchronized void setPartitionLeases(MetastoreContext metastoreContext, S
         throw new UnsupportedOperationException("setPartitionLeases is not supported in FileHiveMetastore");
     }
 
+    @Override
+    public synchronized long lock(MetastoreContext metastoreContext, String databaseName, String tableName)
+    {
+        HiveTableName hiveTableName = hiveTableName(databaseName, tableName);
+        while (lockedHiveTables.containsValue(hiveTableName)) {
+            try {
+                Thread.sleep(10);
```

Comment:
Thanks so much for review! No specific reason actually. Since the `FiveHiveMetastore` only used for test, the sleep here is to mock the waiting of lock acquirement when the lock is occupied by another client or another thread more precisely. `10` is just a polling period which I think is proper.

Reply:
Makes sense to me. Thanks for your clarification.